src/core_plugins/metrics/public/components/vis_editor.js

```diff
@@ -60,27 +69,67 @@ class VisEditor extends Component {
     }
   }
 
-  render() {
-    const handleChange = (part) => {
-      const nextModel = { ...this.state.model, ...part };
-
-      this.props.vis.params = nextModel;
-      if (this.state.autoApply) {
-        this.props.vis.updateState();
-      }
+  isIndexPatternChanged = () => {
+    if (this.state.model.index_pattern !== this.props.vis.params.index_pattern) {
```

Comment:
isnt this doing mostly same stuff as extractIndexPatterns ? could we update this logic to just compare the extracted values to the values on editor state ?

Reply:
mmm no, this logic here just check if the indexPattern in the model is different from configured one in the vis object. `extractIndexPattern` takes the vis object and return an array of configured indexPattern to fetch.
IMHO is better to separate these two concept: get/extract index patterns from the is something changed between the current model and the vis?.
What do you think?

Reply:
forget my previous comment. I need to fix the `extractIndexPattern` that used the `vis.fields` parameter that we removed from the vis object.
Let me check better this method and I will be back with a right implementation